docs/external-node.md

```diff
@@ -0,0 +1,569 @@
+`externalEntitySelector` in the `appliedTo` field. **The `ExternalEntity`
+resource is automatically created for each interface of an `ExternalNode`**,
+which are used by `antrea-controller` to calculate the NetworkPolicies, and
+`antrea-agent` running on the `ExternalNode` receives the internal Antrea
+NetworkPolicy objects.
+
+Following types of (from/to) network peers are supported in an Antrea
+NetworkPolicy applied to an external Node:
+
+- ExternalEntities selected by an `externalEntitySelector`
+- An `ipBlock`
+- A FQDN address in an egress rule
+
+Following actions are supported in an Antrea NetworkPolicy applied to an
+external Node:
+
+- Allow
+- Drop
+- Reject
+
+Below is an example of applying an Antrea NetworkPolicy to the external Nodes
+labeled with `role=db` to reject SSH connections from IP "172.16.100.5" or from
+other external Nodes labeled with `role=front`:
+
+```yaml
+kind: NetworkPolicy
+metadata:
+  name: anp1
+  namespace: vm-ns
+spec:
+  priority: 9000.0
+  appliedTo:
+    - externalEntitySelector:
+        matchLabels:
+          role: db
+  ingress:
+    - action: Reject
+      ports:
+        - protocol: TCP
+          port: 22
+      from:
+        - externalEntitySelector:
+            matchLabels:
+              role: front
+        - ipBlock:
+            cidr: 172.16.100.5/32
+```
+
+### Bypass Antrea NetworkPolicy
+
+In some cases, the special IP traffic is expected to bypass the Antrea
+NetworkPolicy rules on an external Node, e.g.,the SSH connection from a special
+host to the external Node. These rules can be set with the item
+`policyBypassRules` in the agent  configuration. Following is a configuration
+example,
+
+```yaml
+policyBypassRules:
+  - direction: ingress
+    protocol: tcp
+    cidr: 1.1.1.1/32
+    port: 22
+```
+
+The `direction` in one policyBypassRule supports `ingress` or `egress`. And
+the supported protocols include, `tcp`,`udp`, `icmp` and `ip`. The `cidr`
+gives the peer address, it is the destination in an `egress` rule, and the
+source in an `ingress` rule. For `tcp` and `udp` protocols, the `port` is
+required to describe the destination port in the packet.
+
+The policyBypassRules are processed before Antrea NetworkPolicy rules when
+forwarding the packets in OVS datapath, and the disposition of the rules is to
+allow the packets.
+
+## OpenFlow pipeline
+
+A new OpenFlow pipeline is implemented by `antrea-agent` dedicated for
+`ExternalNode` feature.
+
+![OVS pipeline](assets/ovs-pipeline-external-node.svg)
+
+### Non-IP packet
+
+`NonIPTable` is a new OpenFlow table introduced only when on an external Node,
+which is dedicated for all non-IP packets. Non-IP packet is forwarded between
+the pair ports directly, e.g., a non-IP packet entering OVS from the uplink
+interface is output to the paired internal port, and a packet from the internal
+port is output to the uplink.
+
+### IP packet
+
+One OpenFlow pipeline is set on an external Node to process the IP packet.
```

Comment:
"A new OpenFlow pipeline"? Not very sure..

set up

Reply:
updated.